modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -346,6 +347,11 @@ static Query toQuery(QueryShardContext context, boolean mapUnmappedFieldsAsStrin
         // as an analyzed string.
         context.setAllowUnmappedFields(false);
         context.setMapUnmappedFieldAsString(mapUnmappedFieldsAsString);
+        // From 5.0 and onwards the percolator rewrites the query at index time,
+        // this is not the case for percolator queries in indices created before 5.0
+        if (context.getIndexSettings().getIndexVersionCreated().before(Version.V_5_0_0_alpha1)) {
+            queryBuilder = QueryBuilder.rewriteQuery(queryBuilder, context);
```

Comment:
I still think we should rewrite the query all the time. The percolator in 5.x does rewrite at index time but we may have new rewrite rules that should be applied in minor releases ? My point here is that it should be safe to rewrite the query even when the query has been rewritten already.